pkg/data/auth.go

```diff
@@ -3,62 +3,107 @@ package data
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"net/http"
 	"net/url"
-	"regexp"
+	"os"
 	"strings"
 
 	"github.com/dgrijalva/jwt-go"
 )
 
-const KeycloakPath = "http://keycloak:8080"
+const KeycloakPath = "http://keycloak"
 const MicroserviceUserPath = "http://microservice-user:9090"
 
-func SignInRequest(requestBody []byte) []byte {
-	signinPath := KeycloakPath + "/auth/realms/ubivius/protocol/openid-connect/token"
+type Credentials struct {
+    Username string `json:"username"`
+    Password string `json:"password"`
+}
 
-	username := ExtractValue(string(requestBody), "username")
-	password := ExtractValue(string(requestBody), "password")
-	data := url.Values{}
-	data.Set("client_id", "ubivius-client")
-	data.Set("grant_type", "password")
-	data.Set("client_secret", "7d109d2b-524f-4351-bfda-44ecad030eef")
-	data.Set("scope", "openid")
-	data.Set("username", username)
-	data.Set("password", password)
+type TokenResponse struct {
+	AccessToken      string `json:"access_token"`
+	ExpiresIn        int    `json:"expires_in"`
+	RefreshExpiresIn int    `json:"refresh_expires_in"`
+	RefreshToken     string `json:"refresh_token"`
+	TokenType        string `json:"token_type"`
+	NotBeforePolicy  int    `json:"not-before-policy"`
+	SessionState     string `json:"session_state"`
+	Scope            string `json:"scope"`
+}
 
-	//SignIn
-	req, err := http.NewRequest("POST", signinPath, strings.NewReader(data.Encode()))
-	if err != nil {
-		panic(err)
-	}
-	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
-	client := &http.Client{}
-	resp, err := client.Do(req)
-	if err != nil {
-		panic(err)
-	}
-	defer resp.Body.Close()
+type User struct {
+	ID           string `json:"id" bson:"_id"`
+	Username     string `json:"username"`
+	Password	 string `json:"password"`
+	Email        string `json:"email"`
+	FirstName    string `json:"firstname"`
+	LastName     string `json:"lastname"`
+	DateOfBirth  string `json:"dateofbirth"`
+	Gender       string `json:"gender"`
+	Address      string `json:"address"`
+	Bio          string `json:"bio"`
+	Achievements string `json:"achievements"`
+}
 
-	body, _ := ioutil.ReadAll(resp.Body)
-	return body
+type KeycloakUser struct {
+	Username     string `json:"username"`
+	Email        string `json:"email"`
+	FirstName    string `json:"firstName"`
+	LastName     string `json:"lastName"`
+	Enabled      bool	`json:"enabled"`	
 }
 
-func SignUpRequest(requestBody []byte) (string, string) {
-	newUserPath := KeycloakPath + "/auth/admin/realms/ubivius/users"
+/* Claims Map 
```

Comment:
Should this be commented or removed?

Reply:
C'est les claims qu'il y a dans le Token. Par exemple, on va chercher le "sub" pour avoir le ID. Je trouvais intéressant de l'avoir c'est connu ce qui a dans un token, mais ça pourrait être enlevé si tu trouves ça inutile.

Reply:
I think this is interesting but I think I would find a way to document this in the README, as it is general documentation (not contextual).